### DISCUSSION TITLE
Add option to place config inside a directory inside project root directory

### DISCUSSION THREAD

**Discussion by konclave** [evilmartians/lefthook#360] [Ideas]
For exemple place it to `projectDir/config` directory

**Comment by Arkweid** [comment]
I will think how to handle this.

**Comment by lktslionel** [comment]
Hello,

I also think that creating a config file without a dot `.` prefix, break the philosophy of not messing project files with tooling files. Why not following the `.git` directory structure pattern and migrate `lefthook.yml` to `.lefthook/config.yml` ? 
Or at least let the project owner the liberty to define where he want to place lefthook config file.

So, some propositions:
1. Rename `lefthook.yml` to `.lefthook.yml` (As a hidden file by default)
2. Move `lefthook.yml` to `.lefthook/config.yml`
3. Add a new flag `-c|--config` to set the path to a config file for **lefthook**
4. Have so kind of env variable `LEFTHOOK_CONFIG_FILE` or `LEFTHOOK_CONFIG` to override the config file path.

I have ideas about how to implement it. Let me know which proposition sounds great to you, so that I can start working on it and submit a PR soon.

Thanks in advance
Take care
